apis/types.go

```diff
@@ -20,34 +14,21 @@ type Conditioned interface {
 	GetCondition(string) *metav1.Condition
 }
 
-// A V1alpha1ConditionedWithType may have conditions set or retrieved based on agent type. Conditions typically
+// A ConditionedWithType may have conditions set or retrieved based on agent type. Conditions typically
 // indicate the status of both a resource and its reconciliation process.
-type V1alpha1ConditionedWithType interface {
+type ConditionedWithType interface {
```

Comment:
why not move those inside v1alph1 too?

Reply:
The unit test gate started failing 
![image](https://github.com/Azure/fleet/assets/17449572/4643299d-b71c-4080-9e19-ad0380303ef3)

Reply:
Fixed